pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/index/readers/BitSlicedRangeIndexReader.java

```diff
@@ -97,12 +106,22 @@ public ImmutableRoaringBitmap getPartiallyMatchingDocIds(double min, double max)
     return null;
   }
 
-  private ImmutableRoaringBitmap queryRangeBitmap(long min, long max) {
+  private ImmutableRoaringBitmap queryRangeBitmap(long min, long max, long columnMax) {
     RangeBitmap rangeBitmap = mapRangeBitmap();
-    RoaringBitmap lte = rangeBitmap.lte(max);
-    RoaringBitmap gte = rangeBitmap.gte(min);
-    lte.and(gte);
-    return lte.toMutableRoaringBitmap();
+    if (Long.compareUnsigned(max, columnMax) < 0) {
+      RoaringBitmap lte = rangeBitmap.lte(max);
+      if (Long.compareUnsigned(min, 0) > 0) {
```

Comment:
Is this equivalent to `min != 0` (faster)?

Reply:
It's an unsigned comparison, values stored in the index start at 0 and effectively end at -1L. This feels more readable, but checking it's not 0 is probably faster.

Reply:
I don't want to change this because it's less surprising if all of the comparisons use `Long.compareUnsigned` than if only the ones that can't be optimised to something else do. The other operations here will take at least 10s of microseconds even for small segments, so this isn't worth optimising IMO.